code/game/objects/items/tools/medical_tools.dm

```diff
@@ -0,0 +1,246 @@
+
+/obj/item/tool/geltool/gadget/Initialize()
+	. = ..()
+	cell = new /obj/item/tool/geltool/gell()
+
+/obj/item/tool/geltool/gadget/update_icon(mob/user,silent=FALSE)
+	..()
+	if(!cell || cell.gelquantity <= 0)
+		item_state = "synthkitoff"
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+/obj/item/tool/geltool/gadget/proc/gel_use(M = 1)  //change the M if you want to further modify the efficient of a specific procedure
+	cell.gelquantity -= gelefficiency * M
+	usr.visible_message(,"<span class='notice'>The [src] used up [gelefficiency * M] gel in this procedure. [cell.gelquantity] left.</span>")
+
+/obj/item/tool/geltool/gadget/attack_hand(mob/living/user)
+	if(user.get_inactive_held_item() != src)
+		return ..()
+	if(!cell)
+		return ..()
+	cell.update_icon()
+	user.put_in_active_hand(cell)
+	cell = null
+	playsound(user, 'sound/machines/click.ogg', 25, 1, 5)
+	playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+	to_chat(user, "<span class='notice'>You remove the Gell vial from [src].</span>")
+	update_icon(user)
+	return TRUE
+
+/obj/item/tool/geltool/gadget/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(istype(I, /obj/item/tool/geltool/gell))
+		if(!user.drop_held_item())
+			return
+		I.forceMove(src)
+		var/replace_install = "You replace the Gell vial in [src]."
+		if(!cell)
+			replace_install = "You install a Gell vial in [src]."
+		else
+			cell.update_icon()
+			user.put_in_hands(cell)
+		cell = I
+		to_chat(user, "<span class='notice'>[replace_install] <b>Gel quantity: [cell.gelquantity]/[cell.gelcapacity]</b></span>")
+		playsound(user, 'sound/weapons/guns/interact/rifle_reload.ogg', 25, 1, 5)
+		playsound(user, 'sound/weapons/guns/interact/flamethrower_unload.ogg', 12, 2, 5)
+		update_icon()
+
+// TOOLS down here
+/obj/item/tool/geltool/gadget/kitprinter
+	name = "synthetic trauma kit"
+	desc = "A tool that utilizes gel. to act as brute or a burn advanced kit. it have a level to toggle its mode"
+	icon_state = "synthkitbrute"
+	item_state = "synthkitbrute"
+	var/brute = TRUE
+	var/heal_rate = 1.5 //yes better than advanced kits
+	gelefficiency = 56 //same as 5 kits with small cell. same 15 with regular cell
+
+/obj/item/tool/geltool/gadget/kitprinter/update_icon()
+	..()
+	if(!cell || cell.gelquantity <= 0)
+		icon_state = "synthkitoff"
+		return
+	if(brute)
+		icon_state = "synthkitbrute"
+		return
+	icon_state = "synthkitburn"
+
+
+/obj/item/tool/geltool/gadget/kitprinter/proc/update_kitprinter(mob/user, silent=FALSE)
+	if(!cell || cell.gelquantity <= 0)
+		icon_state = "synthkitoff"
+		playsound(loc, 'sound/weapons/saberoff.ogg', 0.25, 1, 2)
+		playsound(loc, 'sound/weapons/guns/misc/empty_alarm.ogg', 25, 1)
+		to_chat(user, "<span class='warning'>The tool beeps as it runs out of gel!</span>")
+		return
+
+	update_icon()
```

Comment:
you're changing the icon stat or updating the icon? icon_state definition should be in update_icon_state, not in other places